ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -671,11 +745,17 @@ private ServerErrorCode validateRequest(PartitionId partition, boolean checkPart
         }
       }
     }
-    // 3. ensure if the partition can be written to
-    if (checkPartitionState && partition.getPartitionState() == PartitionState.READ_ONLY) {
+    // 4. ensure if the partition can be written to
+    if (requestType.equals(RequestOrResponseType.PutRequest)
+        && partition.getPartitionState() == PartitionState.READ_ONLY) {
       metrics.partitionReadOnlyError.inc();
       return ServerErrorCode.Partition_ReadOnly;
     }
+    // 5. Ensure that the request is enabled.
+    if (!isRequestEnabled(requestType, partition)) {
+      metrics.temporarilyUnavailableError.inc();
+      return ServerErrorCode.Temporarily_Unavailable;
```

Comment:
I wanted to keep it generic because I thought it could be reused later when internal state may cause requests to become temp unavailable. I thought that "disabled" is too specific - do you think we should make "disabled" anyway?